lib/cli/cliCompact.js

```diff
@@ -0,0 +1,181 @@
+                        switch (args[1]) {
+                            case 'status':
+                                return this.statusCompactModeInstance(command);
+                            case 'enable':
+                            case 'on':
+                                return this.setCompactModeInstance(command, true, args[2]);
+                            case 'disable':
+                            case 'off':
+                                return this.setCompactModeInstance(command, false);
+                            case 'group':
+                                return this.setCompactModeInstance(command, undefined, args[2]);
+                        }
+                    }
+                }
+
+                messages.error.unknownCommand('compact', command);
+                showHelp();
+                return void callback(3);
+        }
+    }
+
+    /**
+     * Output status of compact mode for host
+     */
+    statusCompactModeHost() {
+        const { callback } = this.options;
+
+        const compactMode = !!(this.config && this.config.system && this.config.system.compact);
+        console.log('Compact mode for this host is currently ' + (compactMode ? 'enabled' : 'disabled'));
+
+        return void callback();
+    }
+
+    /**
+     * Set compact mode for this host
+     * @param targetState target state to set for host
+     */
+    setCompactModeHost(targetState) {
+        const { callback, dbConnect } = this.options;
+
+        dbConnect((_objects, _states, isOffline) => {
+            const compactMode = !!(this.config && this.config.system && this.config.system.compact);
+            console.log('Compact mode for this host is currently ' + (compactMode ? 'enabled' : 'disabled'));
+
+            if (targetState === compactMode) {
+                return void callback();
+            }
+
+            this.config.system = this.config.system || {};
+            this.config.system.compact = targetState;
+            const configFileName = tools.getConfigFileName();
+            fs.writeFileSync(configFileName, JSON.stringify(this.config, null, 4), 'utf8');
+
+            console.log();
+            console.log('Compact mode for this host changed to ' + (targetState ? 'enabled' : 'disabled'));
+
+            if (!isOffline) {
+                console.log();
+                console.log('Please stop ioBroker for this change to take effect!')
+            }
+            return void callback();
+        });
+    }
+
+    /**
+     * Output the compact mode status for a defined adapter instance
+     * @param instance instance to otput status for
+     */
+    statusCompactModeInstance(instance) {
+        const { callback, dbConnect } = this.options;
+
+        dbConnect((objects, _states, isOffline, _objectDbType, config) => {
+            if (!this.config.system || !this.config.system.compact) {
+                console.log('Compact mode is disabled for this host. This setting will have no effect!');
```

Comment:
Which setting do you mean? This only outputs the current status.
How about "This setting currently has no effect!" ?

Reply:
I changed to "You need to enable compact mode for this host for this setting to take effect!"